### DISCUSSION TITLE
nextjs route api and vercel serveless function

### DISCUSSION THREAD

**Discussion by Evan-liu-1018** [vercel/community#4679] [Help]
### Summary

I have a nextjs project and want to deploy it to vercel, but according to the vercel and nextjs related documentation, the nextjs route api and vercel serveless function are the same in code, so I want to know, what is the difference between the two? Will it cause conflicts, I can't find detailed documentation thanks

**Comment by amyegan** [comment]
Hey @Evan-liu-1018! If you're using Node (JavaScript/TypeScript) Serverless Functions then you should use [Route Handlers](https://nextjs.org/docs/app/building-your-application/routing/route-handlers) with the App Router or [API Routes](https://nextjs.org/docs/pages/building-your-application/routing/api-routes) with the Pages Router. When the app is built, Vercel will deploy them as Serverless Functions. 

I recommend setting up all of your Node Serverless Functions in the Next.js way if you're building a Next.js app to avoid routing conflicts.

**Reply by Evan-liu-1018** [reply]
Thank you very much for your answer.I want to know if I write the code completely in the nextjs way and deploy it to vercel, will vercel convert the relevant code into vercel's serveless function?
